pythonAPIClient/client.py

```diff
@@ -45,25 +45,149 @@ def __init__(self, clientId, clientSecret):
 
     # TODO: return clientID and client secret
     def register(self, owner, permissionLevel='user'):
-        pass
+        url = Client.base_URL + 'apps/register/?owner=' + owner + \
+              '&permissionLevel=' + permissionLevel
+        header = self.header_builder()
+        response = requests.post(url, headers=header)
+        if response.status_code == 404:
+            raise ResourceNotFoundError(response.json()['message'])
+        elif response.status_code == 422:
+            raise FormatError(response.json()['message'])
+        elif response.status_code == 401:
+            self.request_token()
+            header = self.header_builder()
+            response = requests.post(url, headers=header)
+        elif response.status_code == 500:
+            raise InternalError()
+        return response.json()['clientID'], response.json()['clientSecret']
+
 
     # TODO: update client's owner
     def update_owner(self, clientID, owner):
-        pass
+        """
+        Parameters from path
+        ----------
+        clientID : str
+            Client Id of the application.
+
+        Parameters from query
+        ----------
+        owner : str
+            (Optional) Username of owner.
+
+        Returns
+        -------
+        str
+            Success message.
+
+        """
+        url = Client.base_URL + 'apps/' + clientID
+        data = {'owner': owner}
+        response = requests.put(url, data)
```

Comment:
I believe you will need to specify header as well. like `requests.put(url, headers=header, data=data)